Texassemble/texassemble.cpp

```diff
@@ -2240,6 +2282,205 @@ int __cdecl wmain(_In_ int argc, _In_z_count_(argc) wchar_t* argv[])
             }
             break;
         }
+
+    case CMD_CUBE_FROM_HC:
+    case CMD_CUBE_FROM_VC:
+    case CMD_CUBE_FROM_VC_FNZ:
+    case CMD_CUBE_FROM_HT:
+    case CMD_CUBE_FROM_HS:
+    case CMD_CUBE_FROM_VS:
+        {
+            auto src = loadedImages.cbegin();
+            auto img = (*src)->GetImage(0, 0, 0);
+            size_t twidth = 0;
+            size_t theight = 0;
+
+            switch (dwCommand)
+            {
+            case CMD_CUBE_FROM_HC:
+            case CMD_CUBE_FROM_HT:
+                twidth = width / 4;
+                theight = height / 3;
+                break;
+
+            case CMD_CUBE_FROM_VC:
+            case CMD_CUBE_FROM_VC_FNZ:
+                twidth = width / 3;
+                theight = height / 4;
+                break;
+
+            case CMD_CUBE_FROM_HS:
+                twidth = width / 6;
+                theight = height;
+                break;
+
+            case CMD_CUBE_FROM_VS:
+                twidth = width;
+                theight = height / 6;
+                break;
+
+            default:
+                break;
+            }
+
+            if (twidth > maxCube || theight > maxCube)
+            {
+                wprintf(L"\nWARNING: Target size exceeds maximum cube dimensions for feature level (%u)\n", maxCube);
+            }
+
+            ScratchImage result;
+            hr = result.InitializeCube(format, twidth, theight, 1, 1);
+            if (FAILED(hr))
+            {
+                wprintf(L"FAILED setting up result image (%08X%ls)\n", static_cast<unsigned int>(hr), GetErrorDesc(hr));
+                return 1;
+            }
+            memset(result.GetPixels(), 0, result.GetPixelsSize());
```

Comment:
You shouldn't need to set it to 0 since you are filling the entirely of each face.

Reply:
It turns out ``InitializeFromCube`` is already zeroing it anyhow I don't need it at line 1965 either :)

Reply:
You mean I should remove all memset lines from this file?
There are 2 memset lines for `Initialize2D`.

Reply:
Anywhere ``memset`` is used right after a ``ScratchImage::Initialize*`` is redundant.